praw/models/reddit/user_subreddit.py

```diff
@@ -0,0 +1,250 @@
+
+    .. note::
+
+        Trying to retrieve attributes of quarantined or private subreddits will result
+        in a 403 error. Trying to retrieve attributes of a banned subreddit will result
+        in a 404 error.
+
+    .. _unix time: https://en.wikipedia.org/wiki/Unix_time
+
+    """
+
+    def __init__(self, reddit: "praw.Reddit", *args, **kwargs):
+        """Initialize a UserSubreddit instance.
+
+        :param reddit: An instance of :class:`~.Reddit`.
+
+        .. note::
+
+            This class should not be initialized directly. Instead obtain an instance
+            via: ``reddit.user.me().subreddit`` or
+            ``reddit.redditor("redditor_name").subreddit``
+
+        """
+
+        def predicate(item):
+            name = getattr(item, "__name__", None)
+            return name not in dir(object) + dir(Subreddit) and name in dir(dict)
+
+        for name, member in inspect.getmembers(dict, predicate=predicate):
+            if name != "__getitem__":
+                setattr(
+                    self,
+                    name,
+                    self._dict_depreciated_wrapper(getattr(self.__dict__, name)),
+                )
+
+        super().__init__(reddit, *args, **kwargs)
+
+    @staticmethod
+    def _dict_depreciated_wrapper(func):
+        """Show deprecation notice for dict only methods."""
+
+        def wrapper(*args, **kwargs):
+            warn(
+                "`Redditor.subreddit` is no longer a dict and is now an `UserSubreddit`"
+                f" object. Using `{func.__name__}` is deprecated and will be removed in PRAW"
+                " 8.",
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    @cachedproperty
+    def mod(self) -> "praw.models.reddit.user_subreddit.UserSubredditModeration":
+        """Provide an instance of :class:`.UserSubredditModeration`.
+
+        For example, to update the authenticated user's display name:
+
+        .. code-block:: python
+
+            reddit.user.me().subreddit.mod.update(title="New display name")
+
+        """
+        return UserSubredditModeration(self)
+
+    def __getitem__(self, item):
+        """Show deprecation notice for dict method `__getitem__`."""
+        warn(
+            "`Redditor.subreddit` is no longer a dict and is now an `UserSubreddit`"
+            " object. Accessing attributes using string indices is deprecated.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        return getattr(self, item)
+
+
+class UserSubredditModeration(SubredditModeration):
+    """Provides a set of moderation functions to a UserSubreddit.
+
+    For example, to accept a moderation invite from subreddit ``r/test``:
+
+    .. code-block:: python
+
+        reddit.subreddit("test").mod.accept_invite()
+
+    """
+
+    def update(
```

Comment:
Is there much difference to this method than the one in SubredditModeration? It's not clear at a glance.

Reply:
Yes, the two uses a different endpoint.